cmd/cmd.go

```diff
@@ -364,6 +364,11 @@ func StartStats() func() {
 
 // initConfig is run by cobra after initialising the flags
 func initConfig() {
+	// Activate logger systemd support if notify socket is detected
+	if socket := os.Getenv("NOTIFY_SOCKET"); socket != "" {
```

Comment:
Are there any other ways of detecting if we are running under systemd?

For example if you were running `rclone serve http` under systemd?

Or maybe those should be supporting the notify socket too?

Potential commands you might run under systemd might be

- serve (http/ftp/sftp/webdav/dlna)
- rcd

Reply:
Yes the trick here is not that we detect that we are running under systemd but that we are running by systemd in notify mode. Notify mode will allows startup and stop signals, status update etc... But this is independent from the logger configuration:
* if we detect the notify socket it means systemd is our parent so we automatically activate the logger systemd support, but:
* we could be launched by systemd in "simple" mode: no notify support, but user could still activate the logger to be compliant with systemd thru the flag. I don't know another way than the notify socket to detect a systemd launch unfortunatly, I will look it up.

So to answer your question (for now): if others commands want to be compatible the notify launch (which is different from just logger support) they will need, like mount, to send the `NotifyReady` signal otherwise systemd will think the start has timed out and kill the process after `ExecStartTimeout`.

Reply:
OK. So the other server commands should probably support the NotifyReady signal...

That sounds like a job for a separate PR!

Reply:
new commits allows to detect sysd launch even without notify support !
see [last comment](https://github.com/rclone/rclone/pull/4609#issuecomment-696155435)